core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/ActivityIntegrationPoint.kt

```diff
@@ -50,14 +51,22 @@ open class ActivityIntegrationPoint private constructor(
     }
 
     companion object {
-        private val integrationPoints = WeakHashMap<Activity, WeakReference<IntegrationPoint>>()
+        private val integrationPoints = HashMap<ComponentActivity, IntegrationPoint>()
 
         fun createIntegrationPoint(
-            activity: Activity,
+            activity: ComponentActivity,
             savedInstanceState: Bundle?,
         ): ActivityIntegrationPoint =
             ActivityIntegrationPoint(activity, savedInstanceState)
-                .also { integrationPoints[activity] = WeakReference(it) }
+                .also {
+                    integrationPoints[activity] = it
+
+                    activity.lifecycle.addObserver(object : DefaultLifecycleObserver {
+                        override fun onDestroy(owner: LifecycleOwner) {
+                            integrationPoints.remove(activity)
```

Comment:
should we remove the observer here?

Reply:
We should run some memory leak analysis on this change to be sure that we are not introducing a memory leak.